spec/api/v1/admin/patients_spec.rb

```diff
@@ -0,0 +1,162 @@
+      before do
+        get patients_crud_url
+      end
+
+      it 'returns all patients' do
+        expect(response).to have_http_status(:ok)
+      end
+    end
+  end
+
+  describe 'GET /api/v1/admin/patients/_id_' do
+    context 'when record exists' do
+      before do
+        patient_url = "#{patients_crud_url}/#{Patient.all.sample.id}"
+        get patient_url
+      end
+
+      it 'return patient' do
+        expect(response).to have_http_status(:ok)
+      end
+    end
+
+    context 'when the request is invalid' do
+      before do
+        get "#{patients_crud_url}/abc"
+      end
+
+      it 'server return error message' do
+        expect(response).to have_http_status(:internal_server_error)
+        expect(JSON.parse(response.body)['error_message']).to eq 'Patient not found'
+      end
+    end
+  end
+
+  describe 'POST /api/v1/admin/patients/' do
+    context 'when params are valid' do
+      before do
+        post patients_crud_url, params: patients_params
+      end
+
+      it 'create new patient and redirect to him' do
+        expect(response).to redirect_to(
+          "#{patients_crud_url}/#{Patient.find_by(passport_id: patients_params[:passport_id]).id}"
+        )
+      end
+    end
+
+    context 'when passport_id has already been used' do
+      before do
+        used_passport_id_params = patients_params
+        used_passport_id_params[:passport_id] = Patient.all.sample.passport_id
+        post patients_crud_url, params: used_passport_id_params
+      end
+
+      it 'server return error message' do
+        expect(response).to have_http_status(:internal_server_error)
+        expect(JSON.parse(response.body)['error_message']).to eq 'Passport has already been taken'
+      end
+    end
+
+    context 'when one of parameters missing' do
+      before do
+        missing_parameters_params = patients_params
+        missing_parameters_params.except!(:full_name)
+        post patients_crud_url, params: missing_parameters_params
+      end
+
+      it 'server return error message' do
+        expect(response).to have_http_status(:bad_request)
+      end
+    end
+  end
+
+  describe 'PUT /api/v1/admin/patients/_id_' do
+    context 'when params are valid' do
+      before do
+        patient = Patient.all.sample
+        put "#{patients_crud_url}/#{patient.id}", params: patients_params
+      end
+
+      it 'update patient and show him' do
+        expect(response).to have_http_status(:ok)
+        expect(JSON.parse(response.body)['passport_id']).to eq patients_params[:passport_id]
+      end
+    end
+
+    context 'when passport_id has already been used' do
+      before do
+        patient = Patient.all.sample
+        used_passport_id_params = patients_params
+        used_passport_id_params[:passport_id] = Patient.all.excluding(patient).sample.passport_id
+        put "#{patients_crud_url}/#{patient.id}", params: used_passport_id_params
+      end
+
+      it 'server return error message' do
```

Comment:
`it 'verb blablabla '`, that's why

`it` = class/module, the stuff you are testing, so you can skip any additional nouns in a description

`it 'returns an error message'`